paddle/majel/malloc.cc

```diff
@@ -0,0 +1,131 @@
+#include "malloc.h"
+#include <glog/logging.h>
+
+#ifndef PADDLE_ONLY_CPU
+#include <cuda_runtime.h>
+#endif
+
+#define CHECK_CUDA(cudaFunc)                                         \
```

Comment:
Yes, but in allocation.cu, we will find

```
Allocation::Allocation(size_t size, Place place) :
    place_(place), size_(size), owned_(true) {

    if (size > 0) {
        majel::detail::Allocator allocator(size_);
        ptr_ = boost::apply_visitor(allocator,
                                    place_);
        if (ptr_ == nullptr) {
            throw std::bad_alloc();
        }
    }
}
```
So, I think that maybe CHECK fatal and throwing a bad_alloc exception will make the same difference. The server will go down, and client can not recover either.